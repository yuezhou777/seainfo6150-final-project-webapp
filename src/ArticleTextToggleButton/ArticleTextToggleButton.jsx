import React from "react";
import style from "./ArticleTextToggleButton.module.css";

const ArticleTextToggleButton = (props) => {
    return (
        <button className={style.button} onClick={props.onClick}>{props.buttonText}</button>
    )
}

export default ArticleTextToggleButton;