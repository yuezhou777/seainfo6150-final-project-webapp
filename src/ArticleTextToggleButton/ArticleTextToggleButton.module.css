.button {
    color: white;
    background-color: #4CAF50;
    height: 35px;
    width: 15em;
    margin-left: 10em;
    border-radius: 5px;
    border: 3px #808080;
    box-shadow: 2px 2px black, -2px -2px gray;
    font-size: 1em;
}