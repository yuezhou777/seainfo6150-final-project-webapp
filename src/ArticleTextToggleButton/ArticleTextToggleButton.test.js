import React from 'react'
import {render} from "@testing-library/react";
import ArticleTextToggleButton from "./ArticleTextToggleButton";

describe('button test', () => {

    it('render ArticleTextToggleButton successfully ', () => {
        const onClick = jest.fn();

        const { container } = render(<ArticleTextToggleButton buttonText={"Show more"} onClick={onClick}/>);
        
        expect(container).toMatchSnapshot();
    });

})