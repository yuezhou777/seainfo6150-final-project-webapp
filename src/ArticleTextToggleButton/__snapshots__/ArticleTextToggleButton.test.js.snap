// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button test render ArticleTextToggleButton successfully  1`] = `
<div>
  <button
    class="button"
  >
    Show more
  </button>
</div>
`;
